### PR TITLE
async generations

### PR DESCRIPTION
**Pull request by jprincipe**
- updated the generation calls to handle each `Cell` asynchronously.
- add `Position` and `Dimensions` struct so the tuples passed around for those values have a structure.

### CODE REVIEW

File: lib/game_of_life/universe/template.ex [hunk 1]
```diff
@@ -1,97 +1,100 @@
...
+      %Position{x: 6, y: 14},
+      %Position{x: 10, y: 14},
+      %Position{x: 11, y: 14},
+      %Position{x: 12, y: 14}
     ]
   end
 
   def initial_state(:penta_decathlon) do
     [
-      {4, 5},
```
Comment (BGRicker) [inline lib/game_of_life/universe/template.ex:73]: definitely like the switch to named arguments here, makes it much easier to read

File: lib/game_of_life_web/live/universe_live.ex [hunk 2]
```diff
@@ -1,7 +1,9 @@
 defmodule GameOfLifeWeb.UniverseLive do
   use Phoenix.LiveView
+
```
Comment (BGRicker) [inline lib/game_of_life_web/live/universe_live.ex:3]: no problem with adding this in but are there usually blank lines between `use` and `alias`es?

**Review by BGRicker — APPROVED** [review]

File: lib/game_of_life_web/live/universe_live.ex (hunk 2 above)
Comment (jprincipe) [reply lib/game_of_life_web/live/universe_live.ex:3]: there isn't a standard for this.  I prefer to separate `import` `use` and `alias` for readability

Comment (kevlarr) [reply lib/game_of_life_web/live/universe_live.ex:3]: I like that

Comment (BGRicker) [reply lib/game_of_life_web/live/universe_live.ex:3]: sounds good to me! I don't have a preference either way, just wanted to make sure it wasn't an oopsy-add

**Pull request merged**